src/neuroconv/tools/nwb_helpers/_configuration_models/_base_backend.py

```diff
@@ -63,4 +67,73 @@ def from_nwbfile(cls, nwbfile: NWBFile) -> Self:
             for default_dataset_configuration in default_dataset_configurations
         }
 
-        return cls(dataset_configurations=dataset_configurations)
+        return cls(dataset_configurations=dataset_configurations, nwbfile_identifier=nwbfile.identifier)
+
+    def is_compatible_with_nwbfile(self, nwbfile: NWBFile) -> bool:
+        """
+        Check if the backend configuration is compatible with a given NWBFile.
+
+        Parameters
+        ----------
+        nwbfile : pynwb.NWBFile
+            The NWBFile object to check compatibility against.
+
+        Returns
+        -------
+        bool
+            True if all NWB object IDs in `nwbfile` are present in this backend configuration's
+            dataset configurations, False otherwise.
+        """
+
+        ids_in_backend_configuration = {
+            dataset_configuration.object_id for dataset_configuration in self.dataset_configurations.values()
+        }
+        ids_in_nwbfile = set(nwbfile.objects.keys())
+        return ids_in_nwbfile.issubset(ids_in_backend_configuration)
+
+    def build_remapped_backend_to_nwbfile(self, nwbfile: NWBFile) -> Self:
+        """
+        Create a new backend configuration remapped to a different NWBFile.
+
+        Parameters
+        ----------
+        nwbfile : pynwb.NWBFile
+            The new NWBFile to remap the configuration to.
+
+        Returns
+        -------
+            A new instance of the backend configuration class, with dataset configurations
+            remapped to their corresponding locations in the new NWBFile.
+
+        Raises
+        ------
+        ValueError
+            If there is a location in the NWBFile that does not have a corresponding
+            configuration in the original configuration.
+
+        Notes
+        -----
+        This function creates a new configuration object. The original configuration remains unchanged.
+        """
+
+        location_to_former_configuration = {
+            dataset_configuration.location_in_file: dataset_configuration
+            for dataset_configuration in self.dataset_configurations.values()
+        }
+
+        backend_configuration_class = type(self)
+        new_backend_configuration = backend_configuration_class.from_nwbfile(nwbfile=nwbfile)
+
+        for dataset_configuration in new_backend_configuration.dataset_configurations.values():
+            location_in_new_nwbfile = dataset_configuration.location_in_file
+            if location_in_new_nwbfile not in location_to_former_configuration:
+                raise ValueError(f"Configuration for object in the following {location_in_new_nwbfile} not found.")
+
+            # Mapping the configuration through locations in the new NWBFile to the former configuration
+            former_configuration = location_to_former_configuration[location_in_new_nwbfile]
+            former_configuration.object_id = dataset_configuration.object_id
```

Comment:
Eh, OK, I see the mutation still happens here

I don't think it's necessary but I can see how it makes the syntax a bit easier on the final setter two lines below

Reply:
It's fine 👍

Reply:
Or maybe not fine depending on if the `former_configuration` is a reference of the source configuration instead of a local copy (which I thought at first)